yapapi/executor/__init__.py

```diff
@@ -124,6 +128,8 @@ def unpack_work_item(item: WorkItem) -> Tuple[Work, ExecOptions]:
 
 
 class Golem(AsyncContextManager):
+    """Base execution engine containing functions common to all modes of operation."""
```

Comment:
Since this has become our main entry point class for the API I'd consider including a bit more information about it here (e.g. the fact it is considered a main entry point, mention it's usually a good idea to have a single instance in the agent code, how and why it can be used as a context manager etc.). This is just a suggestion though, not a change request.

Reply:
yeah, it would be a good idea to include a wider description and an example usage perhaps but indeed, it definitely could be a separate pull request

Reply:
I agree but decided to leave it for a separate PR, as suggested by @shadeofblue

Reply:
A follow-up issue: https://github.com/golemfactory/yapapi/issues/425